### PR TITLE
Improve friends selector

### PR DESCRIPTION
**Pull request by Arden488**
- added another list for "associated users"
- when adding or removing friends - synchronise both lists
- add error messages when no such user exists or you try to add yourself
- added "enter" keydown event

### CODE REVIEW

File: pickee_frontend/src/components/FriendsSelector/FriendsSelector.js [hunk 1]
```diff
@@ -8,34 +8,79 @@ export default {
...
-        removeFriend: function(targetItem) {
-            const targetItemIndex = this.selectedFriends.findIndex(sourceItem => sourceItem.id === targetItem.id);
-            this.selectedFriends.splice(targetItemIndex, 1);
+        removeFriend: function(friend) {
+            this.associatedFriends.find(ascFriend => friend.id === ascFriend.id).hidden = false;
+            const friendIndex = this.selectedFriends.findIndex(sourceItem => sourceItem.id === friend.id);
+            this.selectedFriends.splice(friendIndex, 1);
         },
-        addFriend: async function() {
+        selectFriend: function(friend) {
```
Comment (BelfDev) [inline pickee_frontend/src/components/FriendsSelector/FriendsSelector.js]: "friends" is a bit confusing. Maybe it's a good idea to keep referencing this domain as "associated users"

Comment (Arden488) [reply pickee_frontend/src/components/FriendsSelector/FriendsSelector.js]: ok

Comment (Arden488) [reply pickee_frontend/src/components/FriendsSelector/FriendsSelector.js]: I will rename it in another PR and every place together (there are several places where I use "friend" term). ok?

Comment (BelfDev) [reply pickee_frontend/src/components/FriendsSelector/FriendsSelector.js]: Okay, it's just a suggestion

**Pull request merged**